### ISSUE TITLE
Compilation error in KotlinGenerator generated code when <kotlinNotNullRecordAttributes/> and <recordsGeneratingRecordN/> are both active

### ISSUE THREAD

**Issue by atollk** [jOOQ/jOOQ#14785]
### Expected behavior

By adding the newly introduced (https://github.com/jOOQ/jOOQ/issues/10212) `kotlinNotNullRecordAttributes` configuration property, I expect generated properties to have the null-correct types. For example, the property itself is generated correctly:

```kotlin
    open var id: UUID
        set(value): Unit = set(0, value)
    @Nonnull
        get(): UUID = get(0) as UUID
``` 

(`get()` returns a non-nullable `UUID`)

### Actual behavior

The "valueN" function setters are always nullable and thus cause compile errors due to invalid assignment.

```kotlin
    override fun value1(value: UUID?): UsersRecord {
        this.id = value
        return this
    }
```

### Steps to reproduce the problem

Set up a database with a non-nullable column and, using the Kotlin generator, add `kotlinNotNullRecordAttributes` with value `true` to your "Generate" config. 

[jooq_test.zip](https://github.com/jOOQ/jOOQ/files/10958476/jooq_test.zip)

### jOOQ Version

Community 3.18.0

### Database product and version

Postgres 15

### Java Version

18.0.2 (Amazon.com Inc. 18.0.2+9-FR)

### OS Version

macOS Ventura Version 13.2.1 (22D68)

### JDBC driver name and version (include name if unofficial driver)

org.postgresql:postgresql:42.3.3

**Comment by thomasneuteboom** [comment]
I have the exact same issue at the moment! ☝🏼

**Comment by lukaseder** [comment]
Thanks for your report, and for the reproducer @atollk. I'll have a look at this, soon.

@thomasneuteboom I don't think it's the same issue as you reported: https://github.com/jOOQ/jOOQ/issues/14786

**Comment by lukaseder** [comment]
This particular case seems to be an oversight in the implementation of https://github.com/jOOQ/jOOQ/issues/10212, when `recordsImplementingRecordN` is activated. If you don't need that feature, you can turn it off to prevent the many `value[N]` methods being generated.

I think the solution here should be to call the super class's `AbstractRecrod.set(int, Object)` method instead of the setter on the class itself. The type of the argument of `value[N](T?)` can't be restrained to `T` for the many reasons illustrated in the "bikeshed issue."

**Comment by atollk** [comment]
@lukaseder I suppose simply forcing non-null values at runtime (i.e. `this.id = value!!`) would be an alternative option as well?

**Comment by lukaseder** [comment]
@atollk That was my first thought, but then, I wondered if that's really what people want. After all, the value can be `null` for various reasons despite what users think they want.

Among all the undesirable options, my feeling here is that following the method signature is the least undesirable one. E.g. why would `r.value1(null)` fail, but `r.set(T.COL1, null)` not? Both accept `T?`.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.19.0 and 3.18.1 (https://github.com/jOOQ/jOOQ/issues/14823).

The fix is to call `set(n, value)` directly, instead of going through the setter. This is the most consistent fix that doesn't change any behaviour nor open new cans of worms.
